aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -862,11 +1415,27 @@ module aptos_framework::delegation_pool {
         shareholder: address,
         coins_amount: u64,
         lockup_cycle: ObservedLockupCycle,
-    ): u64 {
-        let inactive_shares = table::borrow_mut(&mut pool.inactive_shares, lockup_cycle);
-        let shares_to_redeem = amount_to_shares_to_redeem(inactive_shares, shareholder, coins_amount);
+    ): u64 acquires GovernanceRecords {
+        let shares_to_redeem = amount_to_shares_to_redeem(
+            table::borrow(&pool.inactive_shares, lockup_cycle),
+            shareholder,
+            coins_amount);
         // silently exit if not a shareholder otherwise redeem would fail with `ESHAREHOLDER_NOT_FOUND`
         if (shares_to_redeem == 0) return 0;
+
+        // Always update governance records before any change to the shares pool.
+        let pool_address = get_pool_address(pool);
+        // <pending inactive shares> of <shareholder> -= <shares_to_redeem>  ---->
+        // <pending inactive shares> of <current voter of shareholder> -= <shares_to_redeem>
+        // no impact on <pending inactive shares> of <next voter of shareholder>
+        if (partial_governance_voting_enabled(pool_address)) {
+            let governance_records = borrow_global_mut<GovernanceRecords>(pool_address);
+            let current_voter = calculate_and_update_delegator_voter_internal(pool, governance_records, shareholder);
+            let current_vote_delegation = update_and_borrow_mut_vote_delegation(pool, governance_records, current_voter);
+            current_vote_delegation.pending_inactive_shares = current_vote_delegation.pending_inactive_shares - shares_to_redeem;
```

Comment:
what if redeeming inactive shares, not pending-inactive ones? pending-inactive voting shares should be decreased only if redeeming pending-inactive shares

Reply:
Good caught! Added a checking on OLC.